### ISSUE TITLE
[Minor] Order of includes matters

### ISSUE THREAD

**Issue by tiolan** [gabime/spdlog#2082]
Using v1.8.1 the order of includes is important.
If I (alphabetically sorted)
```
#include <spdlog/fmt/bin_to_hex.h>
#include <spdlog/spdlog.h>
```
I get `size_t` not defined and a lot other issues.

But if I
```
#include <spdlog/spdlog.h>
#include <spdlog/fmt/bin_to_hex.h>
```
it works fine.

Not sure if this is to be considered a bug, but people tend to say the order of includes should not matter.

**Comment by gabime** [comment]
I think it is already fixed in pr #2035

**Comment by tiolan** [comment]
I just found this PR that I seemed to miss before.
